### PR TITLE
Fix -Wformat violations

### PR DESCRIPTION
**Pull request by am11**
Removes `-Wno-format` by defining formats in minipal/types.h and using it where it's needed to fix violations.

### CODE REVIEW

File: src/native/minipal/types.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (AaronRobinsonMSFT) [inline src/native/minipal/types.h:1]: @LakshanF This is the work I was alluding to regarding `printf`.

**Review by AaronRobinsonMSFT — APPROVED** [review]
:shipit:

Thank you!

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by AaronRobinsonMSFT** [comment]
Failure appears to be known.

File: src/native/minipal/types.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_TYPES_H
+#define HAVE_MINIPAL_TYPES_H
+
+#if defined(TARGET_32BIT) || defined(TARGET_OSX) || defined(TARGET_WINDOWS)
```
Comment (am11) [inline src/native/minipal/types.h:7]: cc @lambdageek we can move these https://github.com/dotnet/runtime/blob/844f6eff7f22b654e413c2c5491b1445f0d1ee13/src/mono/CMakeLists.txt#L774-L791 definitions here with a similar workaround for `HOST_` vs. `TARGET_` https://github.com/dotnet/runtime/blob/ec9fb02a2c6b606ef06acc911a0b104fd3d2a9a3/src/native/libs/System.IO.Compression.Native/extra_libs.cmake#L3

(I just realized that the macros in this file should be prefixed with `MINIPAL_` for consistency)

**Pull request merged**